pkg/kubernetes/templating.go

```diff
@@ -47,9 +47,9 @@ func (cg *ConfigGenerator) OutputAll(w io.Writer) error {
 	buffer := bytes.Buffer{}
 
 	objs := []*IstioResource{
-		getBaseResource(cg.name, handlerKind).spec(cg.handler),
-		getBaseResource(cg.name, instanceKind).spec(cg.instance),
-		getBaseResource(cg.name, ruleKind).spec(cg.rule),
+		getBaseResource(cg.name, cg.namespace, handlerKind).spec(cg.handler),
```

Comment:
can this `cg.namespace` be empty? and if so, would it still work correctly when calling `c.rc.Post().Namespace(inNamespace)`?

Reply:
It can be empty, but we have the constructor which sets the default value to `istio-system`.

The code you refer to is called on an `IstioClient` which wraps a rest client so is not related to this config generator as such. However if no namespace was provided t that function an error would be returned, yes.

Reply:
:+1: